File: frontend/web-app/src/types/investor.ts
export interface Investor {
  investor_id: string;
  investor_name: string;
  investor_type: 'institutional' | 'individual' | 'family_office' | 'pension_fund' | 'endowment' | 'sovereign_wealth' | 'insurance';
  email: string;
  phone?: string;
  address?: Address;
  tax_id?: string;
  kyc_status: 'pending' | 'approved' | 'rejected';
  aml_status: 'pending' | 'approved' | 'rejected';
  accredited_investor: boolean;
  qualified_purchaser: boolean;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface Address {
  street_address: string;
  city: string;
  state: string;
  postal_code: string;
  country: string;
}

export interface InvestorCommitment {
  commitment_id: string;
  investor_id: string;
  fund_id: string;
  commitment_amount: number;
  commitment_date: string;
  commitment_percentage: number;
  called_amount: number;
  uncalled_amount: number;
  distributed_amount: number;
  current_value: number;
  status: 'active' | 'fulfilled' | 'withdrawn';
}

export interface InvestorStatement {
  statement_id: string;
  investor_id: string;
  fund_id: string;
  statement_date: string;
  period_start: string;
  period_end: string;
  beginning_balance: number;
  capital_calls: number;
  distributions: number;
  management_fees: number;
  performance_fees: number;
  other_fees: number;
  unrealized_gains: number;
  realized_gains: number;
  ending_balance: number;
  irr: number;
  moic: number;
}

export interface InvestorDocument {
  document_id: string;
  investor_id: string;
  document_type: 'subscription_agreement' | 'kyc_document' | 'tax_form' | 'statement' | 'notice' | 'other';
  document_name: string;
  file_path: string;
  upload_date: string;
  uploaded_by: string;
  status: 'pending' | 'approved' | 'rejected';
}

export interface CreateInvestorRequest {
  investor_name: string;
  investor_type: Investor['investor_type'];
  email: string;
  phone?: string;
  address?: Address;
  tax_id?: string;
  accredited_investor: boolean;
  qualified_purchaser: boolean;
}

export interface UpdateInvestorRequest extends Partial<CreateInvestorRequest> {
  investor_id: string;
}

export interface CreateCommitmentRequest {
  investor_id: string;
  fund_id: string;
  commitment_amount: number;
  commitment_date: string;
}

File: backend/api-gateway/src/models/investor.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, EmailStr, Field
from enum import Enum


class InvestorType(str, Enum):
    INSTITUTIONAL = "institutional"
    INDIVIDUAL = "individual"
    FAMILY_OFFICE = "family_office"
    PENSION_FUND = "pension_fund"
    ENDOWMENT = "endowment"
    SOVEREIGN_WEALTH = "sovereign_wealth"
    INSURANCE = "insurance"


class Status(str, Enum):
    PENDING = "pending"
    APPROVED = "approved"
    REJECTED = "rejected"


class CommitmentStatus(str, Enum):
    ACTIVE = "active"
    FULFILLED = "fulfilled"
    WITHDRAWN = "withdrawn"


class TransactionType(str, Enum):
    CAPITAL_CALL = "capital_call"
    DISTRIBUTION = "distribution"
    MANAGEMENT_FEE = "management_fee"
    PERFORMANCE_FEE = "performance_fee"
    OTHER = "other"


class TransactionStatus(str, Enum):
    PENDING = "pending"
    PAID = "paid"
    OVERDUE = "overdue"
    CANCELLED = "cancelled"


class Address(BaseModel):
    street_address: str
    city: str
    state: str
    postal_code: str
    country: str


class Investor(BaseModel):
    investor_id: str
    investor_name: str
    investor_type: InvestorType
    email: EmailStr
    phone: Optional[str] = None
    address: Optional[Address] = None
    tax_id: Optional[str] = None
    kyc_status: Status
    aml_status: Status
    accredited_investor: bool
    qualified_purchaser: bool
    is_active: bool = True
    created_at: datetime
    updated_at: datetime


class InvestorCommitment(BaseModel):
    commitment_id: str
    investor_id: str
    fund_id: str
    commitment_amount: float
    commitment_date: datetime
    commitment_percentage: float
    called_amount: float
    uncalled_amount: float
    distributed_amount: float
    current_value: float
    status: CommitmentStatus


class InvestorStatement(BaseModel):
    statement_id: str
    investor_id: str
    fund_id: str
    statement_date: datetime
    period_start: datetime
    period_end: datetime
    beginning_balance: float
    capital_calls: float
    distributions: float
    management_fees: float
    performance_fees: float
    other_fees: float
    unrealized_gains: float
    realized_gains: float
    ending_balance: float
    irr: float
    moic: float


class InvestorDocument(BaseModel):
    document_id: str
    investor_id: str
    document_type: str
    document_name: str
    file_path: str
    upload_date: datetime
    uploaded_by: str
    status: Status


class CreateInvestorRequest(BaseModel):
    investor_name: str = Field(..., min_length=2, max_length=100)
    investor_type: InvestorType
    email: EmailStr
    phone: Optional[str] = None
    address: Optional[Address] = None
    tax_id: Optional[str] = None
    accredited_investor: bool
    qualified_purchaser: bool


class UpdateInvestorRequest(BaseModel):
    investor_id: str
    investor_name: Optional[str] = Field(None, min_length=2, max_length=100)
    investor_type: Optional[InvestorType] = None
    email: Optional[EmailStr] = None
    phone: Optional[str] = None
    address: Optional[Address] = None
    tax_id: Optional[str] = None
    accredited_investor: Optional[bool] = None
    qualified_purchaser: Optional[bool] = None


class CreateCommitmentRequest(BaseModel):
    investor_id: str
    fund_id: str
    commitment_amount: float = Field(..., gt=0)
    commitment_date: datetime


class InvestorTransaction(BaseModel):
    transaction_id: str
    investor_id: str
    fund_id: str
    transaction_type: TransactionType
    amount: float
    transaction_date: datetime
    description: Optional[str] = None
    status: TransactionStatus
    due_date: Optional[datetime] = None
    paid_date: Optional[datetime] = None


class DeleteInvestorResponse(BaseModel):
    message: str
    investor_id: str
    deleted_at: datetime